### PR TITLE
Fix net.readLine infine loop #17534

### PR DESCRIPTION
**Pull request by oXis**
Refers to #17534

### CODE REVIEW

File: lib/pure/net.nim [hunk 1]
```diff
@@ -1542,7 +1542,7 @@ proc readLine*(socket: Socket, line: var string, timeout = -1,
     var c: char
     discard waitFor(socket, waited, timeout, 1, "readLine")
     var n = recv(socket, addr(c), 1)
-    if n < 0: raiseSockError()
+    if n < 0: raiseSockError(); return
```
Comment (dom96) [inline lib/pure/net.nim]: I think you should put this in the `raiseSockError` template

**Review by dom96 — CHANGES_REQUESTED** [review]

File: lib/pure/net.nim (hunk 1 above)
Comment (oXis) [reply lib/pure/net.nim]: I'm pretty new with Nim. If I put it into the template, the `return` will exit `raiseSockError()` and not the `readLine` function? Or is it somehow propagated? (I'll make the chance when I'm back home, later today)

Comment (dom96) [reply lib/pure/net.nim]: Because the template is "dirty" the call to the template gets replaced with its contents

**Comment by oXis** [comment]
I moved the `return` to the template.

**Review by dom96 — APPROVED** [review]
LGTM.

Would be nice to get a test for this as well if you've got the time :)

**Comment by oXis** [comment]
I can try to make a test except I have absolutely no idea how it works :(

**Pull request merged**